### DISCUSSION TITLE
`navigateTo` vs `router.push`

### DISCUSSION THREAD

**Discussion by gbyesiltas** [nuxt/nuxt#22581] [Questions] [answered]
Hey guys, hope all is good

We recently migrated our Nuxt 2 project to Nuxt 3 and as per the documentations, we've migrated our redirects in middlewares and plugins etc. to use `navigateTo` but I'm a bit confused about the `router.push` usage. Do we have conventions or pros/cons to decide between `router.push` and `navigateTo`? 

Thanks!

**Comment by WailanTirajoh** [comment, marked as answer]
code source: https://github1s.com/nuxt/nuxt/blob/main/packages/nuxt/src/app/composables/router.ts#L112-L206

navigateTo is router composable, it do a lot of things we don't have to write ourself. The nice thing is, it's simple & ease to use.

![image](https://github.com/nuxt/nuxt/assets/53980548/0bbcca06-a34f-45f6-81ac-340b0dd80bc2)
It returns router push / replace at the end also.

At the code implementation, it also calls the nuxtHook, where when we need something to do action when the app is redirect / something, we can put it on nuxt hook _(the hook might be deprecated?)_.

**Point to be consider:**
1. Functionality: the navigateTo is on nuxt 3 composable, which is maintain by the core team, and handles navigation like internal route + external url naviation, its reduce the complexity of our code by centralizing navigation logic.
2. Ease of Use: By using navigateTo, we're less likely to make navigation-related mistakes. For example, it ensure that you're navigating to an external URL correctly.
3. Simplified Code: Using navigateTo can lead to cleaner and more concise code, it encapsulates the router.push / router.replace operations & make codebase easier to read & maintain.

**Reply by gbyesiltas** [reply]
Thanks! It looks like it's just preferable to use `navigateTo` in all cases where we can use a composable
